multi-tenancy-observability/README.md

```diff
@@ -0,0 +1,270 @@
+
+Below is a proposal of the configuration of the mapping between tenants and their assigned groups from the active directory:
+
+```yaml
+groups:
+- name: giantswarm-admins
+  tenants:
+  - mc-name
+  - wcdev-name
+  - wcprod-name
+- name: Developers
+  tenants:
+  - wcdev-name
+- name: Ops
+  tenants:
+  - wcdev-name
+  - wcprod-name
+...
+```
+
+The mapping configuration should be dynamically created and updated regardless of the data source by an operator.
+The operator would be able to reconcile any kind of data source and generate the configuration expected by `grafana-multi-tenant-proxy`.
+
+Below is a graph exposing multi-tenancy proposal:
+
+<img src="./assets/future-multi-tenancy.png" width="300" alt="Multi-tenancy proposal on the read path">
+
+#### Operator management
+
+`logging-operator` is an existing component responsible for creating a secret for the multi-tenant-proxy. That secret contains a mapping between credentials and tenant (it's currently equivalent to the cluster id).
+
+We will have to either create a new operator to handle multi-tenancy for all observability tools (`observability-multi-tenancy-operator`) or add this feature into the `logging-operator` and rename it to `observability-operator`.
+
+Atlas' current preference is towards the second option : moving towards having a single, global `observability-operator` which would embed several different controllers, each dedicated to its own observability tool (i.e one controller for logging, another one for multi-tenancy, ...). This approach has the benefit of limiting the number of different operators - and thus related repos - which would not only ease the development process but also make troubleshooting a lot whole easier as one would have only 1 operator to check.
+
+We are open to suggestions on this topic.
+
+### Open questions
+
+#### Teleport issue
+
+We are currently facing an issue with `teleport` authentication:
+[Teleport JWT issue](https://github.com/giantswarm/giantswarm/issues/29719)
+
+We are authenticated on Grafana through a JWT token provided by teleport.
+But that token is not forwarded to the datasource as this is not using the Grafana OAuth plugin to authenticate. Due to this, we cannot access the users groups in the grafana-multi-tenant-proxy.
+
+We are considering a number of options that are not ideal at the moment:
+
+- having double datasource: one for customer with OAuth authentication and another dedicated to GiantSwarm people authenticated throw teleport.
+- having double grafana (same idea than above but for the all grafana instance, not just the datasource).
+- share an authentication secret via cookies.
+- other ideas are welcome!
+
+#### Mapping user's groups to tenants
+
+What interface could we offer our customers to enable them to define the mapping between their groups and their tenants:
+
+- Custom resources like TenantCR, GroupCR
```

Comment:
It is related, but I can see us pulling the concept apart so we can "translate" a project CR to specific custom CRs like a o11ytenant CR or sth. Because Project CR should be depicting project/namespace/team and other info, but here for example we only map OIDC group to o11ytenant, so this is more like RBAC binding.

Reply:
In the beggining, we can have this maintained by us via giant swarm config until we think about the UX side to not block this RFC, but I'm not sure where (should we create a WG around multi-tenancy)?